lief/lief/lief.py

```diff
@@ -0,0 +1,63 @@
+#   Copyright 2014-2019 PUNCH Cyber Analytics Group
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""
+Overview
+========
+
+Parse and abstract PE, ELF and MachO files using LIEF
+
+"""
+
+import json
+import lief
+from typing import Dict, Optional
+from configparser import ConfigParser
+
+from stoq.plugins import WorkerPlugin
+from stoq.exceptions import StoqPluginException
+from stoq import Payload, RequestMeta, WorkerResponse
+
+
+class LiefPlugin(WorkerPlugin):
+    def __init__(self, config: ConfigParser, plugin_opts: Optional[Dict]) -> None:
+        super().__init__(config, plugin_opts)
+
+        if plugin_opts and 'abstract' in plugin_opts:
+            abstract = bool(plugin_opts['abstract'])
+        elif config.has_option('options', 'abstract'):
+            abstract = bool(config.get('options', 'abstract'))
+        self.abstract = abstract
```

Comment:
I would recommend setting a default for all required variables within plugins. In this case, this plugin will raise an exception if `abstract` is not defined within plugin options or the plugin configuration. 

As a side note, I realize several other plugins do not follow this behavior, it's something I plan on fixing in the near future.